src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Why/DependencyGraphFinder.cs

```diff
@@ -31,28 +31,36 @@ internal static class DependencyGraphFinder
+            var runtimeIdentifiers = assetsFile.PackageSpec.RuntimeGraph.Runtimes.Keys
+                .Append(null)
+                .ToList();
             // get all top-level package and project references for the project, categorized by target framework alias
             Dictionary<string, List<string>> topLevelReferencesByFramework = GetTopLevelPackageAndProjectReferences(assetsFile, userInputFrameworks);
 
             if (topLevelReferencesByFramework.Count > 0)
             {
                 foreach (var (targetFrameworkAlias, topLevelReferences) in topLevelReferencesByFramework)
                 {
-                    LockFileTarget? target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: null);
-
-                    // get all package libraries for the framework
-                    IList<LockFileTargetLibrary>? packageLibraries = target?.Libraries;
-
-                    // if the project has a dependency on the target package, get the dependency graph
-                    if (packageLibraries?.Any(l => l?.Name?.Equals(targetPackage, StringComparison.OrdinalIgnoreCase) == true) == true)
-                    {
-                        doesProjectHaveDependencyOnPackage = true;
-                        dependencyGraphPerFramework.Add(targetFrameworkAlias,
-                                                        GetDependencyGraphForTargetPerFramework(topLevelReferences, packageLibraries, targetPackage));
-                    }
-                    else
+                    foreach (var runtimeIdentifier in runtimeIdentifiers)
                     {
-                        dependencyGraphPerFramework.Add(targetFrameworkAlias, null);
+                        LockFileTarget? target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: runtimeIdentifier);
+
+                        // get all package libraries for the framework
+                        IList<LockFileTargetLibrary>? packageLibraries = target?.Libraries;
+
+                        // if the project has a dependency on the target package, get the dependency graph
+                        if (packageLibraries?.Any(l => l?.Name?.Equals(targetPackage, StringComparison.OrdinalIgnoreCase) == true) == true)
+                        {
+                            doesProjectHaveDependencyOnPackage = true;
+                            var targetFrameworkDisplayName = runtimeIdentifier == null ? targetFrameworkAlias : $"{targetFrameworkAlias}/{runtimeIdentifier}";
+                            dependencyGraphPerFramework.Add(targetFrameworkDisplayName,
+                                GetDependencyGraphForTargetPerFramework(topLevelReferences, packageLibraries, targetPackage));
+                        }
+                        else
+                        {
+                            dependencyGraphPerFramework.Add(targetFrameworkAlias, null);
```

Comment:
This is in a for loop with multiple rids, I think it can fail if the package does not exist in any of the rids.

Reply:
Add doesn't allow duplicates. 

If you have 3 rids, and the package is missing in all 3 rids, you'll hit this codepath 3 times and it'll throw when you try to add `targetFrameworkAlias` only instead of `targetFrameworkAlias/rid`

Reply:
oh yeah that should be targetFrameworkDisplayName instead